easydock/run_dock.py

```diff
@@ -213,6 +209,16 @@ def main():
                 del args_dict[arg]
             args.__dict__.update(args_dict)
 
+        has_finished_protonation = check_db_status(args.output, ['smi_protonated', 'source_mol_block_protonated']) and args.protonation
+        has_finished_docking = check_db_status(args.output, ['dock_time'])
+        if  has_finished_protonation or has_finished_docking:
+            print(f'initializing is skipped')
+        else:
+            start_init = time.time()
+            init_db(args.output, args.input, args.ncpu, args.max_stereoisomers, args.prefix)
+            end_init = time.time()
+            print(f'initializing database done in {(end_init - start_init):.2f}s', flush=True)
+
         dask_client = create_dask_client(args.hostfile)
 
         if args.protonation:
```

Comment:
The condition can be extended by `and not has_started_docking` to skip this step